mlpf/pytorch_delphes/LRP/hooks.py

```diff
@@ -0,0 +1,106 @@
+from glob import glob
+import sys, os
+import os.path as osp
+import pickle, math, time, numba, tqdm
+import numpy as np
+import pandas as pd
+import sklearn
+from sklearn.metrics import accuracy_score, confusion_matrix
+import matplotlib, mplhep
+matplotlib.use("Agg")
+import matplotlib.pyplot as plt
+
+#Check if the GPU configuration has been provided
+import torch
+use_gpu = torch.cuda.device_count()>0
+multi_gpu = torch.cuda.device_count()>1
+
+try:
+    if not ("CUDA_VISIBLE_DEVICES" in os.environ):
+        import setGPU
+        if multi_gpu:
+            print('Will use multi_gpu..')
+            print("Let's use", torch.cuda.device_count(), "GPUs!")
+        else:
+            print('Will use single_gpu..')
+except Exception as e:
+    print("Could not import setGPU, running CPU-only")
+
+#define the global base device
+if use_gpu:
+    device = torch.device('cuda:0')
+else:
+    device = torch.device('cpu')
```

Comment:
this seems to be repeated in several places, can you run this GPU detection once from a common file?